BasicRecordingApiKotlin/app/src/main/java/com/google/android/gms/fit/samples/basicrecordingapikotlin/MainActivity.kt

```diff
@@ -0,0 +1,326 @@
+            requestCode?.let {
+                GoogleSignIn.requestPermissions(
+                        this,
+                        requestCode,
+                        getGoogleAccount(), fitnessOptions)
+            }
+        } else signInAction()
+    }
+
+    /**
+     * Handles the callback from the OAuth sign in flow, executing the post sign in function
+     */
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+
+        when (resultCode) {
+            RESULT_OK -> {
+                val postSignInAction = signInResponseMap[requestCode]
+                postSignInAction?.let {
+                    postSignInAction()
+                }
+            }
+            else -> oAuthErrorMsg(requestCode, resultCode)
+        }
+    }
+
+    private fun oAuthErrorMsg(requestCode: Int, resultCode: Int) {
+        val message = """
+            There was an error signing into Fit. Check the troubleshooting section of the README
+            for potential issues.
+            Request code was: $requestCode
+            Result code was: $resultCode
+        """.trimIndent()
+        Log.e(TAG, message)
+    }
+
+    private fun hasOAuthPermissions() = GoogleSignIn.hasPermissions(getGoogleAccount(), fitnessOptions)
+
+    private fun getGoogleAccount() = GoogleSignIn.getAccountForExtension(this, fitnessOptions)
+
+    /**
+     * Subscribes to an available [DataType]. Subscriptions can exist across application
+     * instances (so data is recorded even after the application closes down).  When creating
+     * a new subscription, it may already exist from a previous invocation of this app.  If
+     * the subscription already exists, the method is a no-op.  However, you can check this with
+     * a special success code.
+     */
+    private fun subscribe() {
+        // To create a subscription, invoke the Recording API. As soon as the subscription is
+        // active, fitness data will start recording.
+        // [START subscribe_to_datatype]
+        Fitness.getRecordingClient(this, getGoogleAccount())
+                .subscribe(DataType.TYPE_CALORIES_EXPENDED)
+                .addOnSuccessListener { Log.i(TAG, "Successfully subscribed!") }
+                .addOnFailureListener { Log.i(TAG, "There was a problem subscribing.") }
+        // [END subscribe_to_datatype]
+    }
+
+    /**
+     * Fetches a list of all active subscriptions and log it. Since the logger for this sample
+     * also prints to the screen, we can see what is happening in this way.
+     */
+    private fun dumpSubscriptionsList() {
+        // [START list_current_subscriptions]
+        Fitness.getRecordingClient(this, getGoogleAccount())
+                .listSubscriptions(DataType.TYPE_CALORIES_EXPENDED)
+                .addOnSuccessListener { subscriptions ->
+                    for (sc in subscriptions) {
```

Comment:
can you use a word instead of "sc"? I don't know that means... if it's returning a "subscription", I would go with that. Less guessing for dev makes it easier to read.